File: app/src/main/java/com/carbonplayer/model/network/StreamManager.java
package com.carbonplayer.model.network;

import android.content.Context;

import com.carbonplayer.CarbonPlayerApplication;
import com.carbonplayer.model.entity.SongID;
import com.carbonplayer.model.entity.enums.StreamQuality;
import com.carbonplayer.model.network.entity.StreamingContent;


public class StreamManager {
    private static StreamManager instance;

    public static StreamManager getInstance() {
        if (instance == null) instance = new StreamManager();
        return instance;
    }

    public StreamingContent getStream(Context context, SongID id, String title, boolean download) {
        StreamQuality quality = CarbonPlayerApplication.Companion.getInstance().getPreferences()
                .getPreferredStreamQuality(CarbonPlayerApplication.Companion.getInstance());

        StreamingContent content = new StreamingContent(context, id, title, quality, download);

        return content;

    }
}

File: app/src/main/java/com/carbonplayer/model/network/StreamManager.kt
package com.carbonplayer.model.network

import android.content.Context
import com.carbonplayer.CarbonPlayerApplication
import com.carbonplayer.model.entity.SongID
import com.carbonplayer.model.entity.base.ITrack
import com.carbonplayer.model.network.entity.Stream

object StreamManager {

    fun getStream(context: Context, track: ITrack, download: Boolean): Stream {
        val quality = CarbonPlayerApplication.instance.preferences
                .getPreferredStreamQuality(CarbonPlayerApplication.instance)

        return Stream(context, track, quality, download)

    }
}